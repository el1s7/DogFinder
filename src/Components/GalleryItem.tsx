import { Icon } from 'elementz';
import { useState } from 'react';

import { classNames } from '../Helpers/Functions';

const GalleryItem = function (props: {
	image: string
}) {

	const [preview, setPreview] = useState(false);

	const className = classNames('gallery-item', {
		preview
	})

	return (
		<div
			className={className}>
			<div
				className='gallery-image'
				onClick={() => { setPreview(true); }}>
				<img
					loading='lazy'
					src={props.image}
					alt='A dog'
				/>
			</div>

			<div
				className='preview-container'
				onClick={(e) => {
					e.target === e.currentTarget && setPreview(false);
				}}>
				<img
					src={props.image}
					alt='A dog'
				/>
				<div
					className='preview-close'
					onClick={() => { setPreview(false); }}>
					<Icon name='close' />
				</div>
			</div>

		</div>
	)
};

export default GalleryItem;
