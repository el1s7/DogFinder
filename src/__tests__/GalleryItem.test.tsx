import React from 'react';
import renderer from 'react-test-renderer';

import GalleryItem from './../Components/GalleryItem';

it('renders gallery item correctly', () => {
	expect.hasAssertions();

	const component = renderer
		.create(
			<GalleryItem image='/akita.jpg' />
		)

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

