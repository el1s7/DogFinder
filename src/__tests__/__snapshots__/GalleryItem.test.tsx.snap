// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders gallery item correctly 1`] = `
<div
  className="gallery-item"
>
  <div
    className="gallery-image"
    onClick={[Function]}
  >
    <img
      alt="A dog"
      loading="lazy"
      src="/akita.jpg"
    />
  </div>
  <div
    className="preview-container"
    onClick={[Function]}
  >
    <img
      alt="A dog"
      src="/akita.jpg"
    />
    <div
      className="preview-close"
      onClick={[Function]}
    >
      <i
        className="zwicon-close"
      />
    </div>
  </div>
</div>
`;
